test/auth/sasl-imap-smtp-auth.txt

```diff
@@ -0,0 +1,5 @@
+EHLO mail
```

Comment:
Yes, reusing smtp-auth-login.txt makes sense.

Regarding checks of ENABLE_SASLAUTHD: perhaps we can drop that variable at all and just look at SASLAUTHD_MECHANISMS?